src/writers.ts

```diff
@@ -8,15 +16,29 @@ export function createEvmWriters(indexerName: string) {
 
     const sender = event.args.sender;
     const token = event.args.token;
-    const amount = event.args.amount;
+    const amount = (Number(event.args.amount.toString()) / 1e6).toFixed(2); // Transforms 1990000 to 1.99
     const barcode = event.args.barcode;
 
-    const payment = new Payment(`${sender}/${tx.hash}`, indexerName);
+    const chain = indexerName; // The indexer name corresponds to the chain name
+    let tokenName = getTokenName(token, chain);
+
+    if (!tokenName) {
+      // TODO log missing token
+      tokenName = token;
+    }
+
+    const payment = new Payment(`${tx.hash}`, indexerName);
     payment.sender = sender;
-    payment.token = token;
+    payment.token = tokenName;
     payment.amount = amount;
+
     payment.barcode = barcode;
+    const ipfsData = await getJSON(barcode);
+
     payment.block = block.number;
+    payment.product = ipfsData.type;
+    const paymentDate = new Date(block.timestamp * 1000);
+    payment.date = paymentDate.toDateString();
```

Comment:
Can we store an unix timestamp instead?

Reply:
it's what I had in the first place but then decided date would be useful for users. Reverted back to tt : ba540c239d2f485e2b3b974a3c36ad06bd1efbfb